tests/check_sssd_with_tlog.yml

```
@@ -0,0 +1,27 @@
+---
+- name: Check with-tlog feature exists
+  command: authselect list-features sssd
+  register: __tlog_authselect_features
+  changed_when: false
+
+- name: Check if with-tlog authselect feature is currently enabled
+  command: authselect current
+  register: __tlog_authselect_current
+  changed_when: false
+  failed_when: __tlog_authselect_current.rc not in [0, 2]
+
+- name: Read nsswitch.conf
+  slurp:
+    src: /etc/nsswitch.conf
+  register: __nsswitch_slurp
+
+- name: Decode nsswitch content
+  set_fact:
+    __nsswitch_contents: "{{ __nsswitch_slurp['content'] | b64decode }}"
+
+- name: Check if with tlog authselect feature enabled and nsswitch set correctly
+  assert:
+    that:
+      - __nsswitch_contents | regex_search('^passwd:\\s+sss', multiline=True)
+      - '"with-tlog" in __tlog_authselect_current.stdout'
+  when: '"with-tlog" in __tlog_authselect_features.stdout'
```

```suggestion
  when: '"with-tlog" in __tlog_authselect_features.stdout'
  vars:
    __nsswitch_contents: "{{ __nsswitch_slurp['content'] | b64decode }}"
```

Then you can get rid of the `set_fact` task
In general, it's better to not use `set_fact` since that pollutes the global namespace - better to use `vars` which can be scoped